packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -746,6 +750,8 @@ export function getValidatorApi({
         );
       });
 
+      metrics?.registerValidatorStatuses(currentEpoch, validatorStatuses);
```

Comment:
This is a very odd pattern. IMO we should not trigger status metrics in response to an API call. The status of validators materially change over time not in response to API calls. Would make much more sense to track status after an epoch transition like it's already done in the validator monitor. It's also not necessary for the validator client to be involved at all, since the beacon node already has all required information

Reply:
While that's true the `registerValidator` will be called every epoch. The same pattern has been applied here for other metrics invoked in this file in response to API calls, nevertheless I think the requirements have changed since the original issue so we perhaps don't need the `validatorStatuses` tracked by the [beacon node anymore](https://github.com/ChainSafe/lodestar/pull/5239/files#diff-61206b531bae1c15627a06074c8d264e9d833f3d8f9b12cb6fa65b5cb845fbdcR824) since they are in the [VC metrics](https://github.com/ChainSafe/lodestar/pull/5239/files#diff-6b01906a41fe4b3c383bda4797a4d02c2c81e9c86399fbb408600792e0ec88f0R328)

Reply:
> perhaps don't need the validatorStatuses tracked by the [beacon node anymore](https://github.com/ChainSafe/lodestar/pull/5239/files#diff-61206b531bae1c15627a06074c8d264e9d833f3d8f9b12cb6fa65b5cb845fbdcR824) since they are in the [VC metrics](https://github.com/ChainSafe/lodestar/pull/5239/files#diff-6b01906a41fe4b3c383bda4797a4d02c2c81e9c86399fbb408600792e0ec88f0R328)

We should see the BN and VC as two separate services and not make any assumptions that those will be deployed together.